inference-engine/src/gna_plugin/frontend/scale_factor_calc.hpp

```diff
@@ -443,8 +443,9 @@ class ScaleFactorPerLayer<InferenceEngine::ConcatLayer*> {
         }
 
         // find a source quant value
-        // - 1st candidate - non-activation layer with non-1 scale factor
-        // - 2nd candidate - 1st layer with non-1 scale factor
+        // - 1st candidate - input layer
```

Comment:
We should not requantize inputs I think, the only other solution (maybe even better) would be to take smallest scale factor of all inputs and apply it to make sure that no data is discarded (at cost of accuracy hit)

Reply:
one of the checks before is to check if all inputs have the same scale factor. Currently it is not supported a case when you have 2 inputs with 2 different scale factors. In general, the strategy that you are suggesting is the most reasonable.

Reply:
without this check when there was "const" as first input and "input" as second input then sourceScaleFactor was set to be the const one. So the check was passing but input scale factor was not used.